ui_testing/testcases/basic_tests/test006_orders.py

```diff
@@ -2871,3 +2871,65 @@ def test084_filter_by_changed_by(self):
+        tst_plan1, tst_plan2, article1 = TestPlanAPI().create_multiple_test_plan_with_same_article()
+        tst_plan3, tst_plan4, article2 = TestPlanAPI().create_multiple_test_plan_with_same_article()
+        tst_plan5, tst_plan6, article3 = TestPlanAPI().create_multiple_test_plan_with_same_article()
+        test_units1 = [tst_plan2['testUnits'][0]['name'], tst_plan1['testUnits'][0]['name']]
+        test_units2 = [tst_plan3['testUnits'][0]['name'], tst_plan4['testUnits'][0]['name']]
+        test_units3 = [tst_plan5['testUnits'][0]['name'], tst_plan6['testUnits'][0]['name']]
+        self.info(' Create new order.')
+        self.order_page.create_new_order(article=article1, material_type='Raw Material',
+                                         test_plans=[tst_plan1['testPlan']['text'], tst_plan2['testPlan']['text']],
+                                         set_tstunit= False,save=False)
+        order_no = self.order_page.get_no()
+        self.info('create 3 suborder')
+        self.order_page.create_new_suborder(material_type='Raw Material',
+                                            test_plans=[tst_plan3['testPlan']['text'], tst_plan4['testPlan']['text']],
+                                            article_name=article2, add_tstunit=False)
+        self.order_page.create_new_suborder(material_type='Raw Material',
+                                            test_plans=[tst_plan5['testPlan']['text'], tst_plan6['testPlan']['text']],
+                                            article_name=article3, add_tstunit=False)
+
+        self.order_page.save(save_btn='order:save_btn')
+        order_id = self.order_page.get_order_id()
+        suborders_data, _ = self.orders_api.get_suborder_by_order_id(order_id)
+        suborders = suborders_data['orders']
+        analysis_no = []
+        for suborder in suborders:
+            self.assertEqual(len(suborder['analysis']), 1)
+            analysis_no.append(suborder['analysis'][0])
+        self.assertEqual(3, len(analysis_no))
+        self.info('assert the 3 analysis numbers are triggered for each suborder ')
+        self.assertNotEqual(analysis_no[0], analysis_no[1])
+        self.assertNotEqual(analysis_no[0], analysis_no[2])
+        self.assertNotEqual(analysis_no[1], analysis_no[2])
+        self.orders_page.get_orders_page()
+        self.orders_page.filter_by_order_no(filter_text=order_no)
+        suborders_data = self.order_page.get_child_table_data(index=0)
+        analysis_no_list = []
+        for suborder in suborders_data:
+            analysis_no_list.append(suborder['Analysis No.'])
+        self.assertCountEqual(analysis_no, analysis_no_list)
+        self.orders_page.navigate_to_analysis_active_table()
+        self.analyses_page.filter_by_order_no(filter_text=order_no)
+        analysis_rows = self.order_page.result_table()
+        self.assertEqual(3, len(analysis_rows)-1)
+        analysis_data1 = self.order_page.get_child_table_data(index=2)
```

Comment:
You can use for loop 
For i in range( len(result_table)-1)
Get child data 
Assert

Reply:
done